configurations/dockerDataTemplate.go

```diff
@@ -0,0 +1,124 @@
+package configurations
+
+import (
+	"fmt"
+	"strings"
+)
+
+type instruction interface {
+	WriteInstruction() string
+}
+
+type dockerfileData struct {
+	Stages []stage
+}
+
+type stage []instruction
+
+// Decodes the yaml data and gives the stage instance having all the dockerfile instructions.
+func (s *stage) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var data []interface{}
+	var result []instruction
+	err := unmarshal(&data)
+	if err != nil {
+		return err
+	}
+	*s = append(result, getInstructions(data)...)
```

Comment:
why not just `*s = getInstructions(data)`? Do we need `result`?

Reply:
Yes I agree. `result` has no significance here. In the initial code build, I was modifying the result and then assigning it to `*s`. Later I forgot to get rid of `result`.